File: Data Structures & Algorithms/is-anagram/submission-0.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False

        freq = 26 * [0]
        for i in range(0,len(s)):
            freq[ord(s[i])- ord('a')] += 1
            freq[ord(t[i])- ord('a')] -= 1
        
        for i in range(0,26):
            if(freq[i] != 0):
                return False
        return True

File: Data Structures & Algorithms/hand-of-straights/submission-0.cpp
class Solution {
public:
    bool isNStraightHand(vector<int>& hand, int groupSize) {
        int n = hand.size();
        if(n % groupSize != 0){
            return false;
        }

        // We use map instead of unordered map to store the sorted numbers
        map<int,int> freq;

        // Store the freq of each number in map
        for(int num:hand){
            freq[num]++;
        }

        while(!freq.empty()){
            //Smallest remaining number
            int start = freq.begin()->first;

            for(int x=start; x < start + groupSize; x++){
                if(freq[x] == 0){
                    return false;
                }

                freq[x]--;
                if(freq[x] == 0){
                    freq.erase(x);
                }
            }
        }

        return true;

    }
};